.github: Upload /download large artifacts to s3

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58418 .github: Enable Windows workflow for pull_request
* **#58506 .github: Upload /download large artifacts to s3**

We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

seemethere committed:
.github: Upload /download large artifacts to s3

We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28520792).

zhouzhuojie commented on .github/templates/linux_ci_workflow.yml.in:
```diff
@@ -195,7 +204,7 @@ jobs:
           esac
           echo "SHM_SIZE=${shm_size}" >> "${GITHUB_ENV}"
-      - uses: actions/download-artifact@v2
+      - uses: seemethere/download-artifact-s3@5a1313076ccf40533f94918e2225fe02bd3a96a2
```
nit, publish as a formal version? so that future possible small fix doesn't need to have new PRs again

seemethere replied:
We could definitely do that but I also think that having a commit sha for a non-default action makes it a lot easier to debug things when something goes wrong since it points to the specific commit sha

zhouzhuojie commented on .github/workflows/pytorch-linux-xenial-py3.6-gcc5.4.yml:
```diff
@@ -142,8 +142,17 @@ jobs:
         run: |
           zip -r artifacts.zip dist/ build/
+      # Upload to github so that people can click and download artifacts
       - uses: actions/upload-artifact@v2
```
if this step is fyi only, maybe we can add `continue-on-error: true` to bypass its flakiness

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

seemethere replied:
Done

zhouzhuojie commented on .github/workflows/pytorch-win-vs2019-cpu-py3.yml:
```diff
@@ -42,14 +42,23 @@ jobs:
           .jenkins/pytorch/win-build.sh
-      - name: Upload artifacts
+      # Upload to github so that people can click and download artifacts
+      - name: Upload artifacts to Github
```
same here for `continue-on-error`

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

zhouzhuojie commented on .github/templates/windows_ci_workflow.yml.in:
```diff
@@ -70,7 +81,7 @@ jobs:
         run: |
           .\.github\scripts\install_windows_sdk.ps1
-      - uses: actions/download-artifact@v2
+      - uses: seemethere/download-artifact-s3@0504774707cbc8603d7dca922e8026eb8bf3b47b
```
btw, what's the permission setting for the s3 bucket? Is it public gettable, wondering if we need to run through some security checks?

seemethere replied:
No it's not publicly accessible, I tried adding `public-read` for ACL settings but I still need to fix IAM policies before enabling that

zhouzhuojie reviewed (approved):
lgtm, left a comment about s3 permission

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28520792).

malfet commented on .github/templates/linux_ci_workflow.yml.in:
```diff
@@ -144,8 +144,19 @@ jobs:
+        name: Store PyTorch Build Artifacts on Github
+        with:
+          name: ${{ env.BUILD_ENVIRONMENT }}
+          retention-days: 30
```
Can we reduce retention to just two weeks?

seemethere replied:
Done

malfet commented on .github/templates/windows_ci_workflow.yml.in:
```diff
@@ -44,9 +44,20 @@ jobs:
+        # Don't fail on upload to GH since it's only for user convenience
+        continue-on-error: true
+        with:
+          retention-days: 30
```
Same as above, two weeks should be enough, shouldn't it?

seemethere replied:
Done

seemethere committed:
Update on ".github: Upload /download large artifacts to s3"


We were experiencing 500 errors when it came to downloading large
artifacts so let's just use s3 for those larger artifacts just in case

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28520792](https://our.internmc.facebook.com/intern/diff/D28520792)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28520792).
